src/hardware/configuration.rs

```diff
@@ -211,15 +211,15 @@ pub fn setup(
         // Configure the timer to count at the designed tick rate. We will manually set the
         // period below.
         timer5.pause();
+        timer5.reset_counter();
```

Comment:
Resetting here is unnecessary - the timer is reset immediately before enabling the counter in `lockin-external.rs::init()` during `Timer5::start()`

Reply:
What about Timer2 and Timer3 (e.g. https://github.com/quartiq/stabilizer/blob/e1c87c149f4ed3b76eba343703d5d3879cafb3ae/src/hardware/configuration.rs#L157). Aren't they reset during `start()`ed as well?

There is an underlying issue here: we need to be aware that the timestamping and the sampling timer are not strictly synchronized in their initial counter values. This will lead to a non-deterministic phase shift.

Reply:
Synchronization is not at issue here from my understanding - if we want them to be in sync, that can be done (we do it for the shadow_sampling_timer), but we should do it in a separate PR - this was a design decision that we explicitly made early on.

`timer3`'s reset is necessary because it only receives a "TRIGGER" synchronization input from TIM2 (which starts the counter but does not reset it). However, you are correct that TIM2's reset here is superfluous. In any case, this is a superficial reset that doesn't have an effect on functional firmware, so it's acceptable to leave it in or remove it (user preference).

Reply:
It's good to add that design decision to the code. I have done that now.
And I've removed the spurious TIM2 and TIM5 resets.